metrics/web/test/features/auth/presentation/widgets/auth_form_test.dart

```diff
@@ -0,0 +1,161 @@
+      await tester.pump();
+
+      expect(find.text(AuthStrings.passwordIsRequired), findsOneWidget);
+    });
+
+    testWidgets("password input shows error message if value is less then 6",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+
+      await tester.enterText(passwordInputFinder, '12345');
+      await tester.tap(submitButtonFinder);
+      await tester.pump();
+
+      expect(find.text(AuthStrings.passwordMinLength), findsOneWidget);
+    });
+
+    testWidgets("password input shows a value replaced by bullets",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(const _AuthFormTestbed());
+      await tester.enterText(passwordInputFinder, 'x');
+      await tester.pumpAndSettle(const Duration(seconds: 2));
+
+      final String editText =
+          _findRenderEditable(tester, passwordInputFinder).text.text;
+
+      expect(editText, '\u2022');
+    });
+
+    testWidgets(
+        "signInWithEmailAndPassword method is called on tap on sign in button",
+        (WidgetTester tester) async {
+      final authStore = AuthStoreMock();
+
+      await tester.pumpWidget(_AuthFormTestbed(authStore: authStore));
+      await tester.enterText(emailInputFinder, testEmail);
+      await tester.enterText(passwordInputFinder, testPassword);
+      await tester.tap(submitButtonFinder);
+
+      verify(authStore.signInWithEmailAndPassword(testEmail, testPassword))
+          .called(equals(1));
+    });
+
+    testWidgets("shows an auth error text if the login process went wrong",
+        (WidgetTester tester) async {
+      await tester.pumpWidget(_AuthFormTestbed(
+        authStore: SignInErrorAuthStoreMock(),
+      ));
+      await tester.enterText(emailInputFinder, 'test@email.com');
+      await tester.enterText(passwordInputFinder, 'testPassword');
+      await tester.tap(submitButtonFinder);
+      await tester.pumpAndSettle();
+
+      expect(find.text(SignInErrorAuthStoreMock.errorMessage), findsOneWidget);
+    });
+  });
+}
+
+class _AuthFormTestbed extends StatelessWidget {
+  final AuthStore authStore;
+
+  const _AuthFormTestbed({
+    this.authStore,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Injector(
+      inject: [
+        Inject<AuthStore>(() => authStore ?? AuthStoreMock()),
+      ],
+      initState: () {
+        Injector.getAsReactive<AuthStore>().setState(
+          (store) => store.subscribeToAuthenticationUpdates(),
+        );
+      },
+      builder: (BuildContext context) => MaterialApp(
+        title: 'Auth form',
+        home: Scaffold(
+          body: AuthForm(),
+        ),
+      ),
+    );
+  }
+}
+
+/// Mock implementation of the [AuthStore] that emulates presence of auth message error.
+class SignInErrorAuthStoreMock extends Mock implements AuthStore {
+  static const String errorMessage = "Unknown error";
+
+  SignInErrorAuthStoreMock() {
+    when(authErrorMessage).thenReturn(errorMessage);
+  }
+}
```

Comment:
I think we don't need to create a separate `Mock` to stub a method of this mock. We should use one mock and stub methods in tests if we need it.